doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,291 @@
+| `<CLASS_NAME>`               | yes       | Name of the query handler class                                                                                |
+| `<CLASS_MODULE>`             | yes       | Module name of the query handler class <span style="color: red">(should we mention `builtins` here?)</span>    |
+| `<CLASS_PARAMETERS>`         | yes       | Parameters of the query handler class encoded as string                                                        |
+| `<UDF_NAME>`                 | -         | Name of Python UDF script including user-implemented algorithm                                                 |
+| `<UDF_DB_SCHEMA>`            | -         | Schema name where the UDF script is deployed                                                                   |
+| `<BUCKETFS_CONNECTION_NAME>` | yes       | BucketFS connection name to create temporary file outputs                                                      |
+| `<BUCKETFS_DIRECTORY>`       | yes       | Directory in BucketFS for temporary file outputs                                                               |
+| `<TEMP_DB_SCHEMA>`           | yes       | Database Schema for temporary database objects, e.g. tables                                                    |
+
+# Implementation of Custom Algorithms
+
+Each algorithm should extend the `UDFQueryHandler` abstract class and then implement the following methods:
+* `start()`: This method is called at the first execution of the framework, that is, in the first iteration. It returns a result object: Either _Finish_ or _Continue_.
+  * The _Finish_ result object contains the final result of the implemented algorithm.
+  * The _Continue_ object contains the query list that will be executed for the next state.
+* `handle_query_result()`: This method is called at the following iterations to handle the return query.
+
+Here is an example class definition:
+
+```python
+--/
+CREATE OR REPLACE PYTHON3_AAF SET SCRIPT "MY_SCHEMA"."MY_QUERY_HANDLER_UDF"(...)
+EMITS (outputs VARCHAR(2000000)) AS
+
+from typing import Union
+from exasol_advanced_analytics_framework.udf_framework.udf_query_handler import UDFQueryHandler
+from exasol_advanced_analytics_framework.query_handler.context.query_handler_context import QueryHandlerContext
+from exasol_advanced_analytics_framework.query_result.query_result import QueryResult
+from exasol_advanced_analytics_framework.query_handler.result import Result, Continue, Finish
+from exasol_advanced_analytics_framework.query_handler.query.select_query import SelectQuery, SelectQueryWithColumnDefinition
+from exasol_data_science_utils_python.schema.column import Column
+from exasol_data_science_utils_python.schema.column_name import ColumnName
+from exasol_data_science_utils_python.schema.column_type import ColumnType
+
+
+-- proposal write parameter into temp table in schema ...
```

Comment:
maybe the following scenario

start
1. write parameter+"bucketfs"+time to temp bucketfs file
2. create temp table
3. insert parameter+"table-insert"+time to temp table
4. Return query select from temp table and concat "return"

handle result
1. read result
2. read temp bucketfs file
3. create table from result
4. insert temp bucketfs file content into new table
5. return new table name

Reply:
Proposal: Move to separate PR.